### ISSUE TITLE
There is no scalar implementation for the named  'LocalDateTime' scalar type

### ISSUE THREAD

**Issue by mdevino** [tailrocks/graphql-java-datetime#51]
Hello. Thank you very much for making this library publicly available. I'm trying to use LocalDateTime for a project and I'm getting a message saying there's no scalar named LocalDateTime scalar type, as below:

```
Caused by: graphql.schema.idl.errors.SchemaProblem: errors=[There is no scalar implementation for the named  'LocalDateTime' scalar type]
	at graphql.schema.idl.SchemaGenerator.makeExecutableSchema(SchemaGenerator.java:80) ~[graphql-java-17.3.jar:na]
	at graphql.schema.idl.SchemaGenerator.makeExecutableSchema(SchemaGenerator.java:56) ~[graphql-java-17.3.jar:na]
	at org.springframework.graphql.execution.DefaultGraphQlSourceBuilder.build(DefaultGraphQlSourceBuilder.java:142) ~[spring-graphql-1.0.0-M5.jar:na]
	at org.springframework.boot.autoconfigure.graphql.GraphQlAutoConfiguration.graphQlSource(GraphQlAutoConfiguration.java:88) ~[spring-boot-autoconfigure-2.7.0-M1.jar:2.7.0-M1]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_202]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_202]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_202]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_202]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.3.15.jar:5.3.15]
	... 108 common frames omitted
```

### Step 1

Added the Maven dependency.

```xml
<dependency>
	<groupId>com.zhokhov.graphql</groupId>
	<artifactId>graphql-datetime-spring-boot-starter-webflux</artifactId>
	<version>4.1.0</version>
</dependency>
```

### Step 2

Added `scalar LocalDateTime` to my .graphqls schema file.

## Question

Is there anything I'm missing? I looked through the [README](https://github.com/tailrocks/graphql-java-datetime/blob/master/README.md) and the [WebFlux sample project](https://github.com/tailrocks/graphql-java-datetime/tree/master/graphql-datetime-sample-app-webflux), but couldn't figure it out. If further data is needed, the pet project I'm working on is publicly available [here](https://github.com/mdevino/shiny-timebox/tree/localdatetime).

**Comment by donbeave** [comment]
Hello, @mdevino !

The reason why these starters doesn't work in your project is because you are using official Spring's starter from GraphQL:

```xml
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-graphql</artifactId>
		</dependency>
```

When this library was created at that moment there is no official starter from the Spring team, only a starter from the GraphQL Kickstart project: https://github.com/graphql-java-kickstart

I'm planning to add support for the official Spring Boot GraphQL starter in `5.0.0` version. If you can help me with this, it will be amazing to receive a such PR.

**Comment by mdevino** [comment]
Hi @donbeave. Thanks for the reply. I'm a beginner to the Spring ecossystem as well as GraphQL. I'm not sure on how to make such integration, but I'm willing to try.

I'll take a look at the projects and try to figure out how to get this done and open the PR if I manage to get it working (any directions on where to look, if you'd have them, are welcome).

**Comment by donbeave** [comment]
@mdevino To make it work, you need to find a place where Spring GraphQL is initializing `GraphQLSchema`. I think it will use `SchemaParserBuilder` where you can register scalars by adding them with such method `.scalars(new GraphQLLocalDate())`, you can check this code as an example:

https://github.com/tailrocks/graphql-java-datetime/issues/32#issuecomment-692239086

**Comment by mdevino** [comment]
@donbeave Thank you very much. I'll give it a go.

**Comment by donbeave** [comment]
@mdevino I have added support for Spring GraphQL here: https://github.com/tailrocks/graphql-java-datetime/tree/master/graphql-datetime-spring-boot-starter

It will be released with the new version.

There is a sample application that shows how it works: https://github.com/tailrocks/graphql-java-datetime/tree/master/samples/spring-boot-webmvc

**Comment by mdevino** [comment]
Hi, @donbeave. Thanks for that, and sorry for not getting back to you.
Does [this commit](https://github.com/tailrocks/graphql-java-datetime/commit/8822f81e0a24abfd0dcab72e509ad85e17ac9b85) contains all the code you had to write to make it work with Spring?
Just so I can take a look on what was needed, in case a PR to fix it again in the future might be needed.

**Comment by donbeave** [comment]
@mdevino the basic implementation was here: https://github.com/tailrocks/graphql-java-datetime/commit/e92ad82f1679052dda7932ca49ba57dcf1b6ad28#diff-480bdde79e42752eaabd3eea934260dbfbc8b51825e0b371f497dfd040faa5b5

The next commit was added Spring Boot's configuration https://github.com/tailrocks/graphql-java-datetime/commit/8822f81e0a24abfd0dcab72e509ad85e17ac9b85#diff-312967364a736fd7f317b159835888e048de63d81ad1e2c9b139f51de00f3936

**Comment by mdevino** [comment]
Hey, thanks for the update!

**Comment by donbeave** [comment]
@setchy I'm thinking to publish it with the release of the GraphQL Java Kickstart Spring Boot `13.0.0` version. But I'm not sure when that version will be released.

**Comment by donbeave** [comment]
Released 5.0.0: https://github.com/tailrocks/graphql-java-datetime/releases/tag/v5.0.0
